### DISCUSSION TITLE
MacOS problems! 😭

### DISCUSSION THREAD

**Discussion by JoeSchmo3-007** [MRColorR/money4band#52] [Q&A] [answered]
Hi! So I've done, or at least tried to do, my due diligence in trying to find the answer for myself without making myself look like a fool or ask a stupid question here. However, I'm hitting a wall and I can't figure out what to do next. 

I'm running a Macbook Pro M1 on MacOS 13.4. I have not successfully been able to get the program running...at all. I have a throw away desktop that's running Pop_OS and was able to get through most everything but something with the Shouterrr notifications screwed it up. Now that desktop is no longer usable (My own fault completely unrelated here). SOoooo...all I have to work with is this Macbook Pro.

When I try to execute the script in terminal, I get the following error: 
""MONEY4BAND AUTOMATIC GUIDED SETUP
--------------------------------- 
Checking dependencies...
Now a small useful package named JQ used to manage JSON files will be installed if not already present
Please, if prompted, enter your sudo password to proceed

./runme.sh: line 73: dpkg: command not found
jq is not installed. Trying to install now...
Password:
The operation couldn’t be completed. Unable to locate a Java Runtime that supports apt.
Please visit http://www.java.com for information on installing Java.

Failed to install jq. Please install it manually then restart the script.
Press Enter to exit..."

I randomly came across a proxyrack image in Docker that stated for Mac M1 ARM you needed to change the environment variable in the DOCKER_DEFAULT_PLATFORM to linux_amd64. Apart from setting that as a system wide change...which I don't feel I should do...it didn't help at all and I'm still getting the above error because of Java Runtime. 

(Full disclosure...I'm NO developer/coder/engineer and have only recently really started to teach myself in my free time. So my knowledge is elementary at MOST! 😅) Any ideas what my next step should be? I would run a VM on my machine....but I need the CPU/GPU preserved for other things to function properly.

**Comment by MRColorR** [comment]
Hello, thank you for reaching out.
The information you provided are very useful and i already have an idea on
what to do to fix this. Fixing issue for mac is a bit complicated but i
should be able to work on a fix in the next few weeks.

For now you can proceed following these manual steps:
1)Create accounts (regjster using email/password) for each app youbwnat to
enable/use.
2) install docker downloading the setup from docker desktop website open it
and also check that it will start automatically after a system reboot (if
not chet the docker settings ad enable run on startup).
2) Follow the manual setup guide that you can find in the wiki. ->
Basically edit .env and add the values (email/password/tokens) directly
inside it. Do the same in the docker-compose.yaml file uncommenting each
app that you'd like to use,  removing the starting #ENABLE_APPNAME  comment
for each app.
3) open a terminal in the same folder and use docker compose up -d sto
manually start the project

I'll also post an update message here when the new patch for fixing the
script auto setup for mac will be published.

Il Mar 6 Giu 2023, 12:26 JoeSchmo3-007 ***@***.***> ha
scritto:

**Comment by JoeSchmo3-007** [comment]
Okay! Tried the manual route and thought everything was going well...the docker container was on the dashboard and running, watchtower was also up. But nothing was happening and none of the dashboards for the apps showed that I was connected. Then I realized that I forgot to press save in Visual Studio Code. So I stopped the container and deleted it. Saved the docker-compose.yml file, then attempted to "docker-compose up -d" and now I'm getting the following error:

"parsing /Users/j********/Docker/money4band-v.2.1.0-V2/docker-compose.yaml: yaml: line 6: did not find expected key" 

And...I'm kinda stuck and don't know what is supposed to be at line 6, sometimes it says line 7. I didn't alter anything at the top of that file. Only input the logins that I was supposed to from the .env file.

**Reply by MRColorR** [reply]
ok, just to double check:
- you edited the env file editing the lines with your email/pw or token right ? -> saved .env file
- then you edited the docker compose file just removing the #ENABLE_APPNAME at the start of each line right ? 
  - Are you sure you respected the indentation ? if you remove #ENABLE_APPNAME leaving an extra space or removing too much space the file will break as docker compose expect the file to respect the correct indentation.  -> if not sure try to select the #ENABLE_APPNAME then use the find and replace function of VS code to replace it with an empty string , no spaces. Do it for every specific app for example for #ENABLE_EARNAPP and so on. 
 
If you're willing to test the new automatic setup script for macOS i'm working on you can test the new script downloading the code from this branch https://github.com/MRColorR/money4band/tree/macos-setup-fix . Let me know the test result 👍 
So, if this updated script works fine after testing, I will release it as a pre-release for better testing in the next few weeks and it will then be released in the next stable release along with other features.

**Reply by MRColorR** [reply, marked as answer]
all these changes have been pushed into main and are now part of the latest version
